app/jobs/spotlight_search/export_job.rb

```diff
@@ -0,0 +1,33 @@
+module SpotlightSearch
+  class ExportJob < ApplicationJob
+    def perform(email, records, columns = nil, filters = nil)
+      # data = get_data(records.class, filters)
+      file_path = create_excel(records.as_json, records.class.name)
+      subject = "#{records.class.name&.humanize} export at #{Time.now}"
+      ExportMailer.send_excel_file(email, file_path, subject).deliver_now
+    end
+
+    def get_records(filters)
+      # Todo
+      # Get records based on columns selected and filters
+      # as defined in the model
+    end
+
+    def create_excel(records, klass)
+      # Creating excel with the passed records
+      # Keys as headers and values as row
+      headers = records.first.keys
+      xl = Axlsx::Package.new
+      xl.workbook.add_worksheet do |sheet|
+        sheet.add_row headers, b: true
+        records.map(&:values).each do |values_for_row|
+          sheet.add_row values_for_row
+        end
+        sheet.column_widths 22, 22, 22, 22, 22, 22, 22, 22, 22, 22, 22, 22, 22, 22, 22
```

Comment:
@sajinmp - what does this line do bro? I assume that since the number of columns is dynamic and also we cannot predict the width we will not be able to use this?

Reply:
have changes this. found a way